client/state/login/reducer.js

```diff
@@ -174,29 +174,29 @@ export const socialAccount = createReducer( { isCreating: false, createError: nu
 		bearerToken,
 		createError: null,
 	} ),
-	[ SOCIAL_LOGIN_REQUEST_FAILURE ]: ( state, { error, authInfo } ) => ( {
+	[ SOCIAL_LOGIN_REQUEST_FAILURE ]: ( state, { error } ) => ( {
 		...state,
 		requestError: error,
-		email: error.email,
-		authInfo
 	} ),
 	[ USER_RECEIVE ]: state => ( { ...state, bearerToken: null, username: null, createError: null, } ),
 	[ LOGIN_REQUEST ]: state => ( { ...state, createError: null } ),
 } );
 
-export const socialAccountLink = createReducer( { isLinking: false }, {
-	[ SOCIAL_CREATE_ACCOUNT_REQUEST_FAILURE ]: ( state, { error, token, service } ) => {
-		if ( error.code === 'user_exists' ) {
-			return {
-				isLinking: true,
-				email: error.email,
-				token,
-				service,
-			};
-		}
+const userExistsErrorHandler = ( state, { error, authInfo } ) => {
```

Comment:
Not all calls to the API are within an action creator ( [example](https://github.com/Automattic/wp-calypso/blob/master/client/lib/signup/step-actions.js#L295) ), so I don't think we should rely on `socialInfo` being available on the store and create a dependency to it being available in some previous step.